diff -Nbru st/config.def.h st-sixel/config.def.h
--- st/config.def.h	2021-01-09 16:09:00.392310658 +0000
+++ st-sixel/config.def.h	2021-01-09 15:23:58.481185216 +0000
@@ -206,6 +197,12 @@
 /* Internal keyboard shortcuts. */
 #define MODKEY Mod1Mask
 #define TERMMOD (ControlMask|ShiftMask)
+
+static char *openurlcmd[] = { "/bin/sh", "-c", "st-urlhandler", "externalpipe", NULL };
+
+static char *copyurlcmd[] = { "/bin/sh", "-c", "st-urlhandler -c", "externalpipe", NULL };
+
+static char *copyoutput[] = { "/bin/sh", "-c", "st-copyout", "externalpipe", NULL };
 
 static Shortcut shortcuts[] = {
 	/* mask                 keysym          function        argument */
@@ -223,6 +220,9 @@
 	{ TERMMOD,              XK_Num_Lock,    numlock,        {.i =  0} },
 	{ ShiftMask,            XK_Page_Up,     kscrollup,      {.i = -1} },
 	{ ShiftMask,            XK_Page_Down,   kscrolldown,    {.i = -1} },
+	{ MODKEY,               XK_l,           externalpipe,   {.v = openurlcmd } },
+	{ MODKEY,               XK_y,           externalpipe,   {.v = copyurlcmd } },
+	{ MODKEY,               XK_o,           externalpipe,   {.v = copyoutput } },
 };
 
 /*
diff -Nbru st/Makefile st-sixel/Makefile
--- st/Makefile	2021-01-09 16:08:50.935688158 +0000
+++ st-sixel/Makefile	2021-01-09 00:47:13.040812572 +0000
@@ -44,7 +44,11 @@
 install: st
 	mkdir -p $(DESTDIR)$(PREFIX)/bin
 	cp -f st $(DESTDIR)$(PREFIX)/bin
+	cp -f st-copyout $(DESTDIR)$(PREFIX)/bin
+	cp -f st-urlhandler $(DESTDIR)$(PREFIX)/bin
 	chmod 755 $(DESTDIR)$(PREFIX)/bin/st
+	chmod 755 $(DESTDIR)$(PREFIX)/bin/st-copyout
+	chmod 755 $(DESTDIR)$(PREFIX)/bin/st-urlhandler
 	mkdir -p $(DESTDIR)$(MANPREFIX)/man1
 	sed "s/VERSION/$(VERSION)/g" < st.1 > $(DESTDIR)$(MANPREFIX)/man1/st.1
 	chmod 644 $(DESTDIR)$(MANPREFIX)/man1/st.1
@@ -53,6 +57,8 @@
 
 uninstall:
 	rm -f $(DESTDIR)$(PREFIX)/bin/st
+	rm -f $(DESTDIR)$(PREFIX)/bin/st-copyout
+	rm -f $(DESTDIR)$(PREFIX)/bin/st-urlhandler
 	rm -f $(DESTDIR)$(MANPREFIX)/man1/st.1
 
 .PHONY: all options clean dist install uninstall
diff -Nbru st/st-copyout st-sixel/st-copyout
--- st/st-copyout	1970-01-01 01:00:00.000000000 +0100
+++ st-sixel/st-copyout	2021-01-09 18:24:25.423233200 +0000
@@ -0,0 +1,13 @@
+#!/bin/sh
+# Using external pipe with st, give a dmenu prompt of recent commands,
+# allowing the user to copy the output of one.
+# xclip required for this script.
+# By Jaywalker and Luke
+tmpfile=$(mktemp /tmp/st-cmd-output.XXXXXX)
+trap 'rm "$tmpfile"' 0 1 15
+sed -n "w $tmpfile"
+sed -i 's/\x0//g' "$tmpfile"
+ps1="$(grep "\S" "$tmpfile" | tail -n 1 | sed 's/^\s*//' | cut -d' ' -f1)"
+chosen="$(grep -F "$ps1" "$tmpfile" | sed '$ d' | tac | dmenu -p "Copy which command's output?" -i -c -h 24 -l 20 | sed 's/[^^]/[&]/g; s/\^/\\^/g')"
+eps1="$(echo "$ps1" | sed 's/[^^]/[&]/g; s/\^/\\^/g')"
+awk "/^$chosen$/{p=1;print;next} p&&/$eps1/{p=0};p" "$tmpfile" | xclip -selection clipboard
diff -Nbru st/st-urlhandler st-sixel/st-urlhandler
--- st/st-urlhandler	1970-01-01 01:00:00.000000000 +0100
+++ st-sixel/st-urlhandler	2021-01-09 18:19:05.673252629 +0000
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+clip=0
+verb=Follow
+[ "$1" = "-c" ] && clip=1 && verb=Copy && shift
+
+urlregex="http[s]?:\/\/[a-zA-Z0-9_.~\!*'();:@&=+$,/?#%?-]+"
+
+# First remove linebreaks and mutt sidebars:
+urls="$(sed 's/.*│//g' | tr -d '\n' |
+	grep -aEo "$urlregex" | # grep only urls as defined above.
+#	uniq | # Ignore neighboring duplicates.
+	sed 's/^www./https:\/\/www\./g')"
+
+[ -z "$urls" ] && exit
+
+chosen="$(echo "$urls" | dmenu -i -p "$verb which url?" -c -h 24 -l 10)"
+
+if [ $clip -eq 0 ]; then
+    setsid xdg-open "$chosen" >/dev/null 2>&1 &
+else
+    printf %s "$chosen" | xclip -selection clipboard
+fi
