diff --unified --recursive --text libtsm/src/tsm/tsm-vte.c libtsm-jared/src/tsm/tsm-vte.c
--- libtsm/src/tsm/tsm-vte.c	2020-11-08 15:18:44.655131330 +0000
+++ libtsm-jared/src/tsm/tsm-vte.c	2020-11-08 15:18:16.418761575 +0000
@@ -191,24 +191,24 @@
 
 static uint8_t color_palette[TSM_COLOR_NUM][3] = {
 	[TSM_COLOR_BLACK]         = {   0,   0,   0 }, /* black */
-	[TSM_COLOR_RED]           = { 205,   0,   0 }, /* red */
-	[TSM_COLOR_GREEN]         = {   0, 205,   0 }, /* green */
-	[TSM_COLOR_YELLOW]        = { 205, 205,   0 }, /* yellow */
-	[TSM_COLOR_BLUE]          = {   0,   0, 238 }, /* blue */
-	[TSM_COLOR_MAGENTA]       = { 205,   0, 205 }, /* magenta */
-	[TSM_COLOR_CYAN]          = {   0, 205, 205 }, /* cyan */
-	[TSM_COLOR_LIGHT_GREY]    = { 229, 229, 229 }, /* light grey */
-	[TSM_COLOR_DARK_GREY]     = { 127, 127, 127 }, /* dark grey */
-	[TSM_COLOR_LIGHT_RED]     = { 255,   0,   0 }, /* light red */
-	[TSM_COLOR_LIGHT_GREEN]   = {   0, 255,   0 }, /* light green */
-	[TSM_COLOR_LIGHT_YELLOW]  = { 255, 255,   0 }, /* light yellow */
-	[TSM_COLOR_LIGHT_BLUE]    = {  92,  92, 255 }, /* light blue */
-	[TSM_COLOR_LIGHT_MAGENTA] = { 255,   0, 255 }, /* light magenta */
-	[TSM_COLOR_LIGHT_CYAN]    = {   0, 255, 255 }, /* light cyan */
+	[TSM_COLOR_RED]           = { 237,  21,  21 }, /* red */
+	[TSM_COLOR_GREEN]         = {  17, 209,  22 }, /* green */
+	[TSM_COLOR_YELLOW]        = { 246, 116,   0 }, /* yellow */
+	[TSM_COLOR_BLUE]          = {  29, 153, 243 }, /* blue */
+	[TSM_COLOR_MAGENTA]       = { 155,  89, 182 }, /* magenta */
+	[TSM_COLOR_CYAN]          = {  26, 188, 156 }, /* cyan */
+	[TSM_COLOR_LIGHT_GREY]    = { 239, 239, 239 }, /* light grey */
+	[TSM_COLOR_DARK_GREY]     = { 127, 140, 141 }, /* dark grey */
+	[TSM_COLOR_LIGHT_RED]     = { 192,  57,  43 }, /* light red */
+	[TSM_COLOR_LIGHT_GREEN]   = {  28, 220, 154 }, /* light green */
+	[TSM_COLOR_LIGHT_YELLOW]  = { 253, 188,  75 }, /* light yellow */
+	[TSM_COLOR_LIGHT_BLUE]    = {  61, 174, 233 }, /* light blue */
+	[TSM_COLOR_LIGHT_MAGENTA] = { 142,  68, 173 }, /* light magenta */
+	[TSM_COLOR_LIGHT_CYAN]    = {  22, 160, 133 }, /* light cyan */
 	[TSM_COLOR_WHITE]         = { 255, 255, 255 }, /* white */
 
-	[TSM_COLOR_FOREGROUND]    = { 229, 229, 229 }, /* light grey */
-	[TSM_COLOR_BACKGROUND]    = {   0,   0,   0 }, /* black */
+	[TSM_COLOR_FOREGROUND]    = { 239, 239, 239 }, /* light grey */
+	[TSM_COLOR_BACKGROUND]    = {  35,  38,  39 }, /* black */
 };
 
 static uint8_t color_palette_solarized[TSM_COLOR_NUM][3] = {
